src/assegai/Server.php

```diff
@@ -145,7 +145,7 @@ protected function parsevars(array $s)
 		$this->name        = $this->arrayGet('SERVER_NAME', $s);
 		$this->address     = $this->arrayGet('SERVER_ADDR', $s);
 		$this->port        = $this->arrayGet('SERVER_PORT', $s);
-		$this->remote_addr = $this->arrayGet('HTTP_X_FORWARDED_FOR', $s) ?: $this->arrayGet('REMOTE_ADDR', $s);
+		$this->remote_addr = $this->arrayGet('HTTP_X_FORWARDED_FOR', $s) ?: explode(",", $this->arrayGet('REMOTE_ADDR', $s))[count($s['REMOTE_ADDR'])-1];
```

Comment:
@shaunick sorry, no. it is the `HTTP_X_FORWARDED_FOR` header that you have to split and take the last one of. On production, `REMOTE_ADDR` will be the IP address of the load balancer.

What you want is something like
```php
$this->arrayGet('HTTP_X_FORWARDED_FOR', $s) ? explode(',', $this->arrayGet('HTTP_X_FORWARDED_FOR', $s))[-1] : $this->arrayGet('REMOTE_ADDR', $s);
```

After merging this, we will have to update composer. I will tell you what to do but try to remember.